x-pack/plugins/security_solution/public/detection_engine/rule_management/logic/use_upgrade_security_packages.ts

```diff
@@ -77,7 +77,7 @@ export const useIsUpgradingSecurityPackages = () => {
       const { mutationKey, variables } = options;
 
       // The mutation is bulk Fleet packages installation. Check if the packages include the prebuilt rules package
-      if (mutationKey === BULK_INSTALL_FLEET_PACKAGES_MUTATION_KEY) {
+      if (mutationKey === BULK_INSTALL_FLEET_PACKAGES_MUTATION_KEY && variables) {
```

Comment:
Not sure I understand why this change is needed, for the given mutation variables are required. Do we call it somewhere without arguments?

Reply:
apparently `useIsMutating` is losing the variables for some reason, I think it's related to these changes
https://github.com/TanStack/query/releases/tag/v4.26.1

maybe it would be safer to change the code like below?

```suggestion
      // The mutation is bulk Fleet packages installation. Check if the packages include the prebuilt rules package
      if (mutationKey === BULK_INSTALL_FLEET_PACKAGES_MUTATION_KEY) {
        return (variables as BulkInstallFleetPackagesProps | undefined)
          ? variables.packages.includes(PREBUILT_RULES_PACKAGE_NAME)
          : true;
      }
```

Reply:
@xcrzx what do you think?

Reply:
@patrykkopycinski, sorry, I missed your message. 

My primary concern is that our hook relies on `variables`, and they could potentially be missing. Without checking request params, we can't display the loading indicator. 

I checked locally the update, and it seems we indeed do not show an indicator of an upgrade process happening in the background. The `variables` being `undefined` seems like a bug in React Query to me. How can we identify the root cause? Was this change intentional, or is it a bug? And what are the possible workarounds?

Reply:
@xcrzx, I have changed it to pull the `variables` from the mutation `state` instead of `options` and now it seems to be working as expected